File: base.database/src/main/java/alvin/database/dbflow/converts/LocalDateConvert.java
package alvin.database.dbflow.converts;


import com.google.common.base.Strings;
import com.raizlabs.android.dbflow.converter.TypeConverter;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

public class LocalDateConvert extends TypeConverter<String, LocalDate> {

    @Override
    public String getDBValue(LocalDate model) {
        if (model == null) {
            return null;
        }
        return model.format(DateTimeFormatter.ISO_DATE);
    }

    @Override
    public LocalDate getModelValue(String data) {
        if (Strings.isNullOrEmpty(data)) {
            return null;
        }

        return LocalDate.from(DateTimeFormatter.ISO_DATE.parse(data));
    }
}

File: base.kotlin/src/main/java/alvin/base/kotlin/dbflow/domain/converts/LocalDateConvert.kt
package alvin.base.kotlin.dbflow.domain.converts

import com.raizlabs.android.dbflow.converter.TypeConverter
import java.time.LocalDate
import java.time.format.DateTimeFormatter

class LocalDateConvert : TypeConverter<String, LocalDate>() {

    override fun getModelValue(data: String?): LocalDate? {
        if (data == null) {
            return null
        }
        return LocalDate.from(DateTimeFormatter.ISO_DATE.parse(data))
    }

    override fun getDBValue(model: LocalDate?): String? {
        if (model == null) {
            return null
        }
        return model.format(DateTimeFormatter.ISO_DATE)
    }
}